sections/discussion/scalability-discussion.tex

```diff
@@ -0,0 +1,85 @@
+In this section we will examine and discuss which of these principles we followed throughout the creation of the system.
+
+\begin{itemize}
+    \item \textbf{Design two of everything}: When designing two of everything it is important to identify if there is any single points of failure.
+    In our case that would be the API and the database.
+    For our API we could start multilpe machines that each run an API so that if one fails the others will take over the workload.
+    The same would work for the database, but in that case it would need to handle updating the data of each database instance to ensure consistency.
+    \item \textbf{Scale horizontally}: Horizontal scaling is the smarter way to scale an application, as we talked about in \autoref{sec:scalability}.
+    However we were only able to test it with vertical scaling as running horizontal scaling required us to run the system on multilpe machines and then route the traffic to the different machines.
+    We had difficulties in doing this as we were not able to do this on the university network since each machine gets its own subnetwork.
+    \item \textbf{Build API first}: Whenever we implemented a new functionality we always started with creating the API endpoints.
+    We did this to ensure that the API could be used for other systems in the future and made sure that the data was as generic as possible to not tailor the API for the front end. 
+    \item \textbf{Caching data}: Caching data after retrieval could help increase performance of the system if a user were to quickly swap between separate pages. 
+    As it stands, our system does not take advantage of caching.
+    \item \textbf{Eventual consistency}: The system does not have any parts that strictly require consistency, meaning scalability through availability should be possible. 
+    If a user is browsing the services, it is not imperative that, if a new service were to be added, it would update the page the user was using.
+    It would simply update the next time the user performed an action to change the state of the system, at which point the new data would be loaded and shown.
+    \item \textbf{Easy maintenance and automation}: In order to make maintenance easy, the system has been designed to split responsibility.
+    If an update to the system was necessary, we attempted to make it easy by ensuring it would not require changes in multiple files.
+    \item \textbf{Write asynchronous code}: Allowing tasks to be executed on different threads is a way to increase availability. 
+    This was a primary concern when implementing the system, and all interactions between the API and the database, and the front end and API are processed asynchronously.
+    If the code would not have been asynchronous it would have meant enourmes idle times where the program would have waited for I/O actions to return.
```

Comment:
```suggestion
    If the code would not have been asynchronous it would have meant enormous idle times where the program would have waited for I/O actions to return.
```